podcasts/TranscriptFilter.swift

```diff
@@ -49,9 +60,23 @@ extension RegexFilter {
     // Remove SRT tags, for example: "Speaker 1: " to ""
     static let speakerFilter = RegexFilter(pattern: "Speaker \\d?: *", replacement: "")
     // Ensure that any full stop starts a new line
-    static let fullStopNewLineFilter = RegexFilter(pattern: "([\\!\\?\\.])\\s*", replacement: "$1\n")
+    static let fullStopNewLineFilter = RegexFilter(pattern: "([\\!\\?\\.])\\s+", replacement: "$1\n")
+    // Full Stop at end of cue
+    static let fullStopEndofCueFilter = RegexFilter(pattern: "([\\!\\?\\.])\\z", replacement: "$1\n")
     // Ensure that end of cues have a space when appended to the next cue
     static let notfullStopNewLineFilter = RegexFilter(pattern: "([^\\!\\?\\.])\\z", replacement: "$1 ")
+    // &nbsp filter
+    static let nbspFilter = RegexFilter(pattern: "&nbsp;", replacement: " ")
+    // <br> filter
+    static let breakLineFilter = RegexFilter(pattern: "<br>|<BR>|<br/>|<BR/>", replacement: "\n")
+    // Sound descriptor filter. Ex: [laughs]
+    static let soundDescriptorFilter = RegexFilter(pattern: "\\[[^\\]]*\\]", replacement: "")
+    // Speaker names at start
+    static let htmlSpeakerFilter = RegexFilter(pattern: "^[ ]*\\w+:\\s*", replacement: "")
```

Comment:
I tested with REWORK and it doesn't work there:

<img src="https://github.com/user-attachments/assets/d9f0673d-f1d8-462e-a5b5-5431afc7fc96" width="300">

I feel we'll have many that don't work because HTML is dynamic and everyone can do whatever they want. So as long as we're displaying the actual content right we should be good.

Reply:
I can add some extra filters to remove extra new lines and also to handle the speakers on that format. But as you say, podcasters can format this in any way they feel like.

Reply:
I feel like the extra new lines might be a good one to have? Wdyt?

Reply:
Done here: https://github.com/Automattic/pocket-casts-ios/pull/1941/commits/c72c4c4dd59f89b881cf2b073c9e5e596c7b1d70